docs/book/src/reference/writing-tests.md

```diff
@@ -1,19 +1,33 @@
 # Writing controller tests
 
-Testing Kubernetes controller is a big subject, and the boilerplate testing
+Testing Kubernetes controllers is a big subject, and the boilerplate testing
 files generated for you by kubebuilder are fairly minimal.
 
-[Writing and Running Integration Tests](/reference/testing/envtest.md) documents steps to consider when writing integration steps for your controllers, and available options for configuring your test control plane using [`envtest`](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/envtest).
+To walk you through integration testing patterns for Kubebuilder-generated controllers, we will revisit the CronJob we built in our first tutorial and write a simple test for it.
 
-Until more documentation has been written, your best bet to get started is to look at some
-existing examples, such as:
+The basic approach is that, in your generated `suite_test.go` file, you will use envtest to create a local Kubernetes API server, instantiate and run your controllers, and then write additional `*_test.go` files to test it using [Ginko](http://onsi.github.io/ginkgo).
+
+If you want to tinker with how your envtest cluster is configured, see section [Writing and Running Integration Tests](/reference/testing/envtest.md) as well as the [`envtest docs`](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/envtest).
+
+## Test Environment Setup
+
+{{#literatego ../cronjob-tutorial/testdata/project/controllers/suite_test.go}}
+
+## Testing your Controller's Behavior
```

Comment:
instead of putting this under reference, why don't we just make it a page under the cronjob tutorial?